File: frontend/src/types/availability.ts
export interface SkillTag {
  id: string
  employee_id: string
  skill_name: string
  proficiency: "beginner" | "intermediate" | "advanced" | "expert"
  added_by: string
  added_at: string
  notes?: string
}

export interface SkillCatalogEntry {
  id: string
  name: string
  category: string
  display_name: string
}

export interface ProjectRef {
  project_id: string
  project_name: string
  status: string
  role: string
  end_date?: string | null
  client_name?: string | null
  /** YYYY-MM period string — shown as "last active: Jan 2026" in the table */
  period?: string | null
}

export interface AvailableEmployee {
  employee_id: string
  employee_name: string
  designation: string
  department: string
  location: string
  skills: SkillTag[]
  utilisation_percent: number
  classification: string
  available_from?: string
  /** Active / on-hold projects (backward compat alias for active_projects) */
  current_projects: ProjectRef[]
  active_projects: ProjectRef[]
  /** Last completed projects — shows why the person is on bench */
  last_projects: ProjectRef[]
  /** Date the last project ended (ISO date string) */
  bench_since?: string | null
  /** Days since bench_since */
  bench_duration_days?: number | null
}

export interface BenchPoolResponse {
  period?: string | null
  data_source?: "hrms" | "excel"
  employees: AvailableEmployee[]
  total: number
  bench_count: number
  partial_count: number
  avg_bench_days?: number | null
}

File: backend/app/schemas/availability.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class SkillTagRequest(BaseModel):
    skill_name: str
    proficiency: str = "intermediate"  # beginner | intermediate | advanced | expert
    notes: Optional[str] = None


class SkillTagResponse(BaseModel):
    id: str
    employee_id: str
    skill_name: str
    proficiency: str
    added_by: str
    added_at: datetime
    notes: Optional[str] = None


class SkillCatalogEntry(BaseModel):
    id: str
    name: str
    category: str
    display_name: str


class AvailableEmployee(BaseModel):
    employee_id: str
    employee_name: str
    designation: str
    department: str
    location: str
    skills: list[SkillTagResponse]
    utilisation_percent: float
    classification: str
    available_from: Optional[str] = None
    # Active/on-hold project assignments (same as current_projects for backward compat)
    current_projects: list[dict]
    active_projects: list[dict] = []
    # Last completed projects (for bench context)
    last_projects: list[dict] = []
    # When the employee became bench (end date of their last completed project)
    bench_since: Optional[str] = None
    # Days on bench since bench_since
    bench_duration_days: Optional[int] = None


class BenchPoolResponse(BaseModel):
    employees: list[AvailableEmployee]
    total: int
    bench_count: int
    partial_count: int
    avg_bench_days: Optional[int] = None
